src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -112,36 +124,93 @@ void ColumnIsNullTableScanImpl::_scan_value_segment(const BaseValueSegment& segm
     return;
   }
 
-  DebugAssert(segment.is_nullable(), "Columns that are not nullable should have been caught by edge case handling.");
+  DebugAssert(segment.unique_values_count() != 0 && segment.unique_values_count() != segment.size(),
+              "Columns without or with exclusivly NULLs should have been caught by edge case handling.");
 
-  auto iterable = NullValueVectorIterable{segment.null_values()};
+  auto iterable = create_iterable_from_attribute_vector(segment);
 
-  const auto invert = _predicate_condition == PredicateCondition::IsNotNull;
+  const auto invert = predicate_condition == PredicateCondition::IsNotNull;
   const auto functor = [&](const auto& value) {
     return invert ^ value.is_null();
   };
-  iterable.with_iterators([&](auto iter, auto end) {
+
+  iterable.with_iterators(position_filter, [&](auto iter, auto end) {
     _scan_with_iterators<false>(functor, iter, end, chunk_id, matches);
   });
 }
 
-bool ColumnIsNullTableScanImpl::_matches_all(const BaseValueSegment& segment) const {
-  switch (_predicate_condition) {
+void ColumnIsNullTableScanImpl::_scan_null_value_vector(const std::optional<pmr_vector<bool>>& null_values,
```

Comment:
Just a thought: The methods for dictionary segment and segments with NULL value vector are also almost identical – only the iterable is different. We can reduce code duplication even more by using templating w/ segment type as I sketched before.

Reply:
As discussed in Slack, we mix things here. Catalog information (is a table attribute nullable) and segment information (does it contain NULLs).
For immutable encoded segments, we should have something like `contains_nulls()` as we can determine that when encoding.

For ValueSegments however, the segment information `is_nullable()` is effectively the same as catalogue informatioin, as we need to set the NULL vector whenever a column is nullable (because there might be NULLs added at some point).

So the current interface makes somewhat sense (although it should be properly documented) for the ValueSegments. Having `is_nullable()` for encoded segements, does not sound right to me though.

Reply:
Did we agree on keeping the interfaces as is?

Resolve?

Reply:
We did, but I nevertheless I tried to archive at least some harmonization:
`DictionarySegments`, `LZ4Segments`, and `FrameOfReferenceSegments` now own a method `contains_nulls()` which checks if their `_null_values` std::optional is set.
Moreover, the `null_values()` method of `LZ4Segments` and `FrameOfReferenceSegments` now asserts that `contains_nulls()` is true and returns a reference to their NULL vector instead of the optional itself.
The necessary changes to the interables were not too big, and with this harmonization, we can unify the handling of `DictionarySegments`, `LZ4Segments`, and `FrameOfReferenceSegments` in the `ColumnIsNullTableScan`.
What do you think of these changes?